### ISSUE TITLE
[UX] Improve app exception/mute selection and allow ignoring specific crashes

### ISSUE THREAD

**Issue by rugk** [TacoTheDank/Scoop#56]
## Currently

You have a big checklist of all apps that are installed… and randomly in the alphabetically sorted list the ones you muted are checked.

#### Problems

* Finding an app is hard. ONe needs to navigate and think of the alphabet to find an app.
* Knowing what you have (already) muted is hard, given it is somewhere in that big list

## Improvments

IMHO this would improve the situation:
* A way (button) to whitelist an app or even a specific crash (IMHO much better).
* A search feature to find the apps/a specific app (in 99% of the cases I want to do that when I am on that page, I hardly care about all installed apps)
* Listing the disabled ones first. Under the IMHO realistic assumption that most apps are not muted, and only few are muted, this allows one to easily disable the apps.

Less important IMHO would be:
* Multi-select to disable/enable multiple apps at the same time.

## Comparsion

E.g. look how the modern settings in Android 12 e.g. handle app-related stuff, like the app list., an interesting feature there is:
* sorting by size (IMHO not needed here as it is irrelevant for us(
* a screen before showing only the most recent apps and a button "Show all apps" -> actually a good idea, but not about usage, but maybe by the count of crashes recorded. Generally it may be sorted by "most crashed apps first", so actually that may already show the app I want to mute first.
* a filter button to hide/show system apps (could be quite useful IMHO)

The "app notifications" settings screen alos has some interesting UX aspects:
* It has a filter with which one can filter and sort apps based on status, e.g. "last used", "most frequently" (could be useful if we again take it as the number of crashes) and disabled apps.

## Use cases/User stories
* As a user of the app, I may want to ignore specific crashes, because one crash is well-known, reported and happens way too often (interrupts my workflow/annoying notification).
   * As a user of the app, I also want to quickly find and unmute apps, that I have muted, because if the reason was a reported crash and it is fixed now, I can continue collecting crashes. (Alternative solution: Hiding specific crashes only as said.)
* Actually I have not encountered another reason/use case for ignoring an app, actually…

Of course all that may need some testing, these are only ideas.

**Comment by TacoTheDank** [comment]
>Finding an app is hard. One needs to navigate and think of the alphabet to find an app.

Remembering the alphabet sure is difficult. 😉 I can't really see how it is an app issue to find an app by the alphabet? Unless that's meant to be applied to the following...

>Knowing what you have (already) muted is hard, given it is somewhere in that big list

Good point. I should have the list be organized to put the blacklisted items first (or have an option to do so).

>A way (button) to whitelist an app

This is technically duplicate functionality that IDK if I'd want to implement. But I'll think about it.

>even a specific crash (IMHO much better).

Interesting concept. I'll keep it on the shelf for now.

>A search feature to find the apps/a specific app (in 99% of the cases I want to do that when I am on that page, I hardly care about all installed apps)

There is the magnifying glass (search icon) in the top right, so that already exists.

>Listing the disabled ones first. Under the IMHO realistic assumption that most apps are not muted, and only few are muted, this allows one to easily disable the apps.

Yeah, this is definitely a good idea. It would better behavior for the disabled apps to be put first.

>a screen before showing only the most recent apps and a button "Show all apps" (...) It has a filter with which one can filter and sort apps based on status, e.g. "last used", "most frequently" (could be useful if we again take it as the number of crashes) and disabled apps.

I'm pretty sure that would require special permissions for usage data access. I wanna keep permissions for Scoop as minimal as possible, so I'll probably say no to that.

Might add other stuff, I've not much time ATM.

**Comment by rugk** [comment]
> I can't really see how it is an app issue to find an app by the alphabet? 

Well… scrolling though a list of 1000 apps is… Alphabet is better than unsorted, but I've given examples for even cleverer sorting e.g.

An example where alphabet sorting is superior to search is e.g. if you only know the "last" part of an app, i.e. e.g. a word not a the beginning, you cannot find it without a search or so. 

BTW feel free to split this pile of Ideas into more concrete/separate issues, if you want… :upside_down_face:

**Comment by rugk** [comment]
> I'm pretty sure that would require special permissions for usage data access. I wanna keep permissions for Scoop as minimal as possible, so I'll probably say no to that.

I agree and it does, indeed. So yeah, by "number of crashes/exceptions" is anyway the more useful sorting and it's data you – theoretically – already have.
